core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
Good question!

I checked where `allow_blank` could be coming from and found out this commit:
https://github.com/spree/spree/commit/4c723135e69577606ef2379ef60422e3711c718c

I thought of leaving it there would be the best solution. But I didn't go into too much details. What do you think?